Sources/Identity.swift

```diff
@@ -42,15 +43,58 @@ import Foundation
     }
 
     public func readyForEvent(_ event: Event) -> Bool {
+        if event.urlVariables, getSharedState(extensionName: IdentityConstants.SharedState.Configuration.SHARED_OWNER_NAME, event: event)?.value == nil { return false}
+
         return state.bootupIfReady(getSharedState: getSharedState(extensionName:event:),
                                    createXDMSharedState: createXDMSharedState(data:event:))
     }
 
     // MARK: Event Listeners
 
-    /// Handles events requesting for identifiers. Dispatches response event containing the identifiers. Called by listener registered with event hub.
+    /// Handles events requesting for identifiers. Called by listener registered with event hub.
     /// - Parameter event: the identity request event
     private func handleIdentityRequest(event: Event) {
+        if event.urlVariables {
+            processGetUrlVariablesRequest(event: event)
+        } else {
+            processGetIdentifiersRequest(event: event)
+        }
+    }
+
+    /// Handles events requesting for url variables. Dispatches response event containing the url variables string.
+    /// - Parameter event: the identity request event
+    func processGetUrlVariablesRequest(event: Event) {
+        guard let configurationSharedState = getSharedState(extensionName: IdentityConstants.SharedState.Configuration.SHARED_OWNER_NAME, event: event)?.value else {
+            Log.warning(label: friendlyName, "\(#function) - Cannot process getUrlVariables request Identity event, configuration not found.")
+            return
+        }
+
+        // if config doesn't have org id we cannot proceed.
+        guard let orgId = configurationSharedState[IdentityConstants.ConfigurationKeys.EXPERIENCE_CLOUD_ORGID] as? String, !orgId.isEmpty else {
+            Log.warning(label: friendlyName, "\(#function) - Cannot process getUrlVariables request Identity event, experienceCloud.org is invalid or missing in configuration.")
+            return
```

Comment:
It looks like for all these warning use-cases we return without dispatching an event which will then result in a callbackTimeout error. We should consider dispatching an event with empty response url variables and convert that into an unexpected error in the public api class